src/test/java/com/amplitude/EventTest.java

```diff
@@ -0,0 +1,45 @@
+package com.amplitude;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+public class EventTest {
+  @Test
+  public void testCreateEventWithNullUserAndDeviceThrowsException() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> {
+          new Event("test event", null, null);
+        },
+        "Event must have one defined userId and/or deviceId");
+  }
+
+  @Test
+  public void testToJsonObject() {
+    Event event = new Event("test event", "test-user");
+    String longStr =
+        "This is a long string that is too long for an event, and going to be truncated later when we send the event.";
```

Comment:
Might have to revisit this one. The test is correct, however, there will be different behavior for batch API vs normal API. I believe the Batch API only starts truncating at much longer strings. Also maybe use code that more easily shows the length of this string.

Reply:
Right now I don't think there is any logic in event is concerning if it is batch api or normal api though. And I thought the difference between batch and normal apis is more on how many events we can send with api.

I guess I can make this shorter then assert the length before the truncate runs, which should be easier to see the difference.

Reply:
Yeah, that's the issue I was getting at. The HTTP V2 API vs Batch API treats this differently. Now that we support batch API, does this complicate the tests, and does it require new code behavior?

So the normal HTTP V2 API doesn't allow strings longer than 1024 characters, in its documentation. I don't see a limit in Batch API, only a file size of 20MB per payload. I swear I saw a different limit for Batch API somewhere.

Reply:
Yeah, for now the event logic and truncate logic is separate with the api though, and we are relying on the api response now to tell. If we have a know document about this, then we can decide if we want to adjust the truncate logic based on the api type. For the test now, we can leave for it now